examples/lang_err_integration_tests/lib.rs

```diff
@@ -0,0 +1,284 @@
+                Err(_) => {
+                    unimplemented!("No other `LangError` variants exist at the moment.")
+                }
+            }
+        }
+
+        /// Returns the current value of the `LangErrIntegrationTests`'s boolean.
+        #[ink(message)]
+        pub fn get(&self) -> bool {
+            self.value
+        }
+
+        /// Flips the current value of the `LangErrIntegrationTests`'s boolean.
+        #[ink(message)]
+        pub fn flip(&mut self) {
+            self.value = !self.value;
+        }
+
+        /// Flips the current value of the `LangErrIntegrationTests`'s boolean.
+        ///
+        /// We should see the state being reverted here, no write should occur.
+        #[ink(message)]
+        #[allow(clippy::result_unit_err)]
+        pub fn err_flip(&mut self) -> Result<(), ()> {
+            self.flip();
+            Err(())
+        }
+    }
+
+    #[cfg(test)]
+    mod e2e_tests {
+        type E2EResult<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+        #[ink_e2e::test(additional_contracts = "../flipper/Cargo.toml")]
+        async fn e2e_invalid_selector_can_be_handled(
+            mut client: ink_e2e::Client<C, E>,
+        ) -> E2EResult<()> {
+            use lang_err_integration_tests::contract_types::ink_primitives::{
+                types::AccountId as E2EAccountId,
+                LangError as E2ELangError,
+            };
+
+            let constructor = lang_err_integration_tests::constructors::new();
+            let contract_acc_id = client
+                .instantiate(&mut ink_e2e::charlie(), constructor, 0, None)
+                .await
+                .expect("instantiate failed")
+                .account_id;
+
+            let flipper_constructor = flipper::constructors::default();
+            let flipper_acc_id = client
+                .instantiate(&mut ink_e2e::charlie(), flipper_constructor, 0, None)
+                .await
+                .expect("instantiate `flipper` failed")
+                .account_id;
+
+            let get_call_result = client
+                .call(
+                    &mut ink_e2e::charlie(),
+                    flipper_acc_id.clone(),
+                    flipper::messages::get(),
+                    0,
+                    None,
+                )
+                .await
+                .expect("Calling `flipper::get` failed");
+            let initial_value = get_call_result
+                .value
+                .expect("Input is valid, call must not fail.");
+
+            let flipper_ink_acc_id = E2EAccountId(flipper_acc_id.clone().into());
+            let invalid_selector = [0x00, 0x00, 0x00, 0x00];
+            let call_result = client
+                .call(
+                    &mut ink_e2e::charlie(),
+                    contract_acc_id.clone(),
+                    lang_err_integration_tests::messages::call(
+                        flipper_ink_acc_id,
+                        invalid_selector,
```

Comment:
This tests the failure case but we don't test the success case? We could modify the `e2e_can_flip_correctly` test to flip via `lang_err_integration_tests::messages::call`

Reply:
I figured there wasn't really a need for it since we check for it here:

https://github.com/paritytech/ink/blob/476ed6515aeffc04f92d4dbd73aa1fec1dcf61b2/examples/lang_err_integration_tests/lib.rs#L203-L206

It's only the failure case that needs to be tested using the `CallBuilder` since we can't craft an invalid message using the E2E tests.